### DISCUSSION TITLE
v0.7.1

### DISCUSSION THREAD

**Discussion by laipz8200** [langgenius/dify#7415] [Releases]
## ✨ What’s New in v0.7.1? ✨

Hey everyone, we’re excited to bring you version 0.7.1 of our platform! This release is packed with new features, enhancements, and crucial fixes. Let’s dive in:

### 🚀 New Features

- **Custom App Icons**: Personalize your apps with custom icons by @xuzuodong in [#7196](https://github.com/langgenius/dify/pull/7196).
  
  <img width="769" alt="Screenshot 2024-08-19 at 2 36 27 PM" src="https://github.com/user-attachments/assets/56fb1dd5-2580-4cde-ac6e-01ca036093fd">

- **OpenAI Structured Outputs Support**: Now supporting OpenAI's Structured Outputs by @hjlarry in [#7258](https://github.com/langgenius/dify/pull/7258).
- **Dify Sandbox v0.2.6**: Install any Python package you want by @Yeuoly in [#7410](https://github.com/langgenius/dify/pull/7410).

### 🧠 Model Updates

- **ChatGPT-4o-Latest**: Added the latest ChatGPT-4o model by @hjlarry in [#7289](https://github.com/langgenius/dify/pull/7289).

### 🛠️ New Tools

- **Jina Tokenizer Tool**: Added a new tokenizer tool by @hjlarry in [#7375](https://github.com/langgenius/dify/pull/7375).
- **GitLab Tool**: Integrated GitLab tool by @wlrnet in [#7329](https://github.com/langgenius/dify/pull/7329).

### ⚙️ Enhancements

- **Xinference Auth System**: Enhanced support for Xinference's authentication system by @ethan-fly in [#7369](https://github.com/langgenius/dify/pull/7369).
- **Ruff Formatter**: Introduced Ruff Formatter for better code formatting by @laipz8200 in [#7291](https://github.com/langgenius/dify/pull/7291).
- **Ruff Update**: Updated Ruff from 0.5.x to 0.6.x by @bowenliang123 in [#7384](https://github.com/langgenius/dify/pull/7384).
- **Rearranged API Python Dependencies** by @bowenliang123 in [#7391](https://github.com/langgenius/dify/pull/7391)

### 🛠️ Bug Fixes

- **Rerank Mode Fix**: Fixed missing rerank_mode when converting to DatasetEntity by @optionals in [#7269](https://github.com/langgenius/dify/pull/7269).
- **Tool Test Modal**: Fixed issue where parameters couldn’t be input in tool test modal by @wellCh4n in [#7281](https://github.com/langgenius/dify/pull/7281).
- **Markdown Conversion**: Fixed file to markdown conversion issue by @laipz8200 in [#7293](https://github.com/langgenius/dify/pull/7293).
- **Image Resizing**: Dify editor now supports image resizing by @crazywoola in [#7296](https://github.com/langgenius/dify/pull/7296).
- **Conversation Variables**: Added conversation variables in app DSL service by @laipz8200 in [#7304](https://github.com/langgenius/dify/pull/7304).
- **Node and Edge Issues**: Fixed inability to add node and change the edge by @aixgeek in [#7303](https://github.com/langgenius/dify/pull/7303).
- **Default Values**: Fixed missing default value of type array object in conversation variable modal by @JzoNgKVO in [#7309](https://github.com/langgenius/dify/pull/7309).
- **Elasticsearch Docker Env**: Fixed docker environment for Elasticsearch by @QIN2DIM in [#7270](https://github.com/langgenius/dify/pull/7270).
- **Run History**: Corrected iteration times number in run history by @iamjoel in [#7318](https://github.com/langgenius/dify/pull/7318).
- **Keywords Fix**: Fixed keywords issue by @crazywoola in [#7357](https://github.com/langgenius/dify/pull/7357).
- **Workflow Variables**: Added conversation variable to workflow by @laipz8200 in [#7257](https://github.com/langgenius/dify/pull/7257).

### 🌐 Internationalization

- **Japanese Translation**: Added JP translation for variable aggregator by @HanqingZ in [#7277](https://github.com/langgenius/dify/pull/7277).
- **Fixes in Japanese Translations**: Corrected translations of secret values by @yukyu30 in [#7279](https://github.com/langgenius/dify/pull/7279).
- **General i18n Updates**: Various internationalization updates by @ZuzooVn in [#7312](https://github.com/langgenius/dify/pull/7312).

That’s all for this release! As always, thank you for your contributions and feedback. Keep building and innovating! 

---

### Docker compose deployments

> [!WARNING]
> The `docker-compose.yaml` has been refactored. If you've made any changes to the file, make sure to check out the "Upgrade to new docker compose deployment" section above for usage and migration tips.

1. Back up your customized docker-compose YAML file (optional)

   ```bash
   cd docker
   cp docker-compose.yaml docker-compose.yaml.$(date +%s).bak
   ```

2. Get the latest code from the main branch

   ```bash
   git checkout main
   git pull origin main
   ```

3. Stop the service，Command, please execute in the docker directory

   ```bash
   docker compose down
   ```

4. Back up data

   ```bash
   tar -cvf volumes-$(date +%s).tgz volumes
   ```

5. Upgrade services

   ```bash
   docker compose up -d
   ```

### Source Code deployments

1. Stop API server, Worker and Web frontend Server.

2. Get the latest code from the main branch:

   ```bash
   git checkout main
   git pull origin main
   ```

3. Update Python dependencies:

   ```bash
   cd api
   poetry install
   ```

7. Then, let's run the migration script:

   ```bash
   poetry shell
   flask db upgrade
   ```

8. Finally, run API server, Worker and Web frontend Server again.

---

## What's Changed
* fix: missed rerank_mode when convert to DatasetEntity by @optionals in https://github.com/langgenius/dify/pull/7269
* fix: typo in upstage/llm/_position.yaml by @GareArc in https://github.com/langgenius/dify/pull/7286
* fix ja-JP translation of secret values by @yukyu30 in https://github.com/langgenius/dify/pull/7279
* fix: #7222 docstrings by @ZuzooVn in https://github.com/langgenius/dify/pull/7276
* Add jp translation for variable aggregator by @HanqingZ in https://github.com/langgenius/dify/pull/7277
* fix: can not input param value in tool test modal by @wellCh4n in https://github.com/langgenius/dify/pull/7281
* feat(api/workflow): Add `Conversation.dialogue_count` by @laipz8200 in https://github.com/langgenius/dify/pull/7275
* feat: add chatgpt-4o-latest by @hjlarry in https://github.com/langgenius/dify/pull/7289
* feat: support OPENAI json_schema by @hjlarry in https://github.com/langgenius/dify/pull/7258
* chore(api/core): Improve FileVar's type hint and imports. by @laipz8200 in https://github.com/langgenius/dify/pull/7290
* chore(api): Introduce Ruff Formatter. by @laipz8200 in https://github.com/langgenius/dify/pull/7291
* feat: support pinning, including, and excluding for Model Providers and Tools by @GareArc in https://github.com/langgenius/dify/pull/7283
* fix(api/core/app/segments/segments.py): Fix file to markdown. by @laipz8200 in https://github.com/langgenius/dify/pull/7293
* Feat/7023 dify editor can resize the image by @crazywoola in https://github.com/langgenius/dify/pull/7296
* fix(api/services/app_dsl_service.py): Add conversation variables. by @laipz8200 in https://github.com/langgenius/dify/pull/7304
* fix: inability-to-add-node-and-change-the-edge by @aixgeek in https://github.com/langgenius/dify/pull/7303
* Fix: missing default value of type array object in conversation variable modal by @JzoNgKVO in https://github.com/langgenius/dify/pull/7309
* chore(api/libs): Apply ruff format. by @laipz8200 in https://github.com/langgenius/dify/pull/7301
* fix(elasticsearch): docker env by @QIN2DIM in https://github.com/langgenius/dify/pull/7270
* chore: #7222 i18n by @ZuzooVn in https://github.com/langgenius/dify/pull/7312
* fix: null annotation by @ZuzooVn in https://github.com/langgenius/dify/pull/7313
* add some api  to DifyClient by @lingfengchencn in https://github.com/langgenius/dify/pull/7314
* chore: delete unused resources POSTGRES_MAX_CONNECTIONS by @ZuzooVn in https://github.com/langgenius/dify/pull/7315
* Update PerfXCloud Model List by @BingGeX in https://github.com/langgenius/dify/pull/7212
* fix error when use farui-plus model by @oiuv in https://github.com/langgenius/dify/pull/7316
* fix editor auth by @cuckootan in https://github.com/langgenius/dify/pull/7297
* fix: not show correct iteration times number in run history  by @iamjoel in https://github.com/langgenius/dify/pull/7318
* enh:setfocus after voice input by @leslie2046 in https://github.com/langgenius/dify/pull/7317
* Revert "feat: support pinning, including, and excluding for Model Providers and Tools" by @GareArc in https://github.com/langgenius/dify/pull/7324
* chore(api/libs/bearer_data_source.py): Remove expired fie. by @laipz8200 in https://github.com/langgenius/dify/pull/7300
* feat: show path variable friendly in tool edit by @wellCh4n in https://github.com/langgenius/dify/pull/7344
* chore: update docstrings by @ZuzooVn in https://github.com/langgenius/dify/pull/7343
* feat: tools/gitlab by @wlrnet in https://github.com/langgenius/dify/pull/7329
* add workflowClient ,fix rename bug by @lingfengchencn in https://github.com/langgenius/dify/pull/7352
* feat: support siliconflow rerank by @Weaxs in https://github.com/langgenius/dify/pull/7337
* fix: keywords by @crazywoola in https://github.com/langgenius/dify/pull/7357
* add base url for moonshot model by @oiuv in https://github.com/langgenius/dify/pull/7360
* fix(api/services/workflow/workflow_converter.py): Add converrsation variable to workflow. by @laipz8200 in https://github.com/langgenius/dify/pull/7257
* chore: add LOG_FILE to docker-compose by @ZuzooVn in https://github.com/langgenius/dify/pull/7372
* chore: optimize ark model parameters by @sinomoe in https://github.com/langgenius/dify/pull/7378
* feat(api): support wenxin text embedding by @CheneyYin in https://github.com/langgenius/dify/pull/7377
* feat: add jina tokenizer tool by @hjlarry in https://github.com/langgenius/dify/pull/7375
* feat: custom app icon by @xuzuodong in https://github.com/langgenius/dify/pull/7196
* chore: update ruff from 0.5.x to 0.6.x by @bowenliang123 in https://github.com/langgenius/dify/pull/7384
* feat: support xinference's auth system by @ethan-fly in https://github.com/langgenius/dify/pull/7369
* chore: rearrange api python dependencies by @bowenliang123 in https://github.com/langgenius/dify/pull/7391
* Feat: shortcut hook by @YIXIAO0 in https://github.com/langgenius/dify/pull/7385
* fix:start of the period should be YYYY-MM-DD 00:00 by @leslie2046 in https://github.com/langgenius/dify/pull/7371
* fix(api/core/model_runtime/model_providers/__base/large_language_model.py): Add TEXT type checker by @laipz8200 in https://github.com/langgenius/dify/pull/7407
* feat: OpenRouter add gpt-4o-2024-08-06 model by @XiaoLey in https://github.com/langgenius/dify/pull/7409
* add CrossRef builtin tool: doi query and title query by @Sakura4036 in https://github.com/langgenius/dify/pull/7406
* upgrade: sandbox to 0.2.5 by @Yeuoly in https://github.com/langgenius/dify/pull/7410
* chore(*): Bump version 0.7.1 by @laipz8200 in https://github.com/langgenius/dify/pull/7389

## New Contributors
* @optionals made their first contribution in https://github.com/langgenius/dify/pull/7269
* @GareArc made their first contribution in https://github.com/langgenius/dify/pull/7286
* @yukyu30 made their first contribution in https://github.com/langgenius/dify/pull/7279
* @wellCh4n made their first contribution in https://github.com/langgenius/dify/pull/7281
* @BingGeX made their first contribution in https://github.com/langgenius/dify/pull/7212
* @CheneyYin made their first contribution in https://github.com/langgenius/dify/pull/7377
* @ethan-fly made their first contribution in https://github.com/langgenius/dify/pull/7369

**Full Changelog**: https://github.com/langgenius/dify/compare/0.7.0...0.7.1

<hr /><em>This discussion was created from the release <a href='https://github.com/langgenius/dify/releases/tag/0.7.1'>v0.7.1</a>.</em>

**Comment by Azem92** [comment]
[]()

**Comment by PNjobs** [comment]
I have updated to latest version using source code deployments steps but do not see the variable assigner node nor the conversation variables. When I check About it show v 0.7.2. What am I missing?

**Reply by laipz8200** [reply]
Please check if you are using a chatflow(not workflow).

**Reply by PNjobs** [reply]
Yes I am building a chatflow and I would like to utilize the latest conversation variables and thats why upgraded from version 6 to 7

**Reply by PNjobs** [reply]
I do not see the latest examples "patient intake" and other 2 chatflows that shows usage of conversation variable too after the update. I have deployed the source code way

**Reply by crazywoola** [reply]
> I do not see the latest examples "patient intake" and other 2 chatflows that shows usage of conversation variable too after the update. I have deployed the source code way

What's your chosen language? You can switch to English for this template.

**Reply by laipz8200** [reply]
> Yes I am building a chatflow and I would like to utilize the latest conversation variables and thats why upgraded from version 6 to 7
> 
> 

Could you provide detailed deployment steps along with screenshots for each step? Alternatively, you could try redeploying according to the documentation and check if the issue still persists.

**Reply by PNjobs** [reply]
I followed step 1 to 5 mentioned in this thread about source code deployment. I have done the same steps at least 4 - 5 times. But apart from the version in about i do not see the changes that is expected in 0.7.2 

![dify](https://github.com/user-attachments/assets/bc2e5035-ca56-420c-afd8-eb31368880e3)

**Reply by PNjobs** [reply]
> > I do not see the latest examples "patient intake" and other 2 chatflows that shows usage of conversation variable too after the update. I have deployed the source code way
> 
> What's your chosen language? You can switch to English for this template.

Its English. I mean i do not see the updates of version .7.2
